app/charts/map/map-state.tsx

```diff
@@ -173,13 +174,20 @@ const useMapState = ({
 
       // Right now hierarchies are only created for geoShapes
       if (isGeoShapesDimension(dimension) && geoShapes) {
+        const dimensionLabels = dimension.values.map((d) => d.label);
         const hierarchyLabels = (
           (geoShapes as any).topology.objects.shapes.geometries as GeoFeature[]
         )
           .filter((d) => d.properties.hierarchyLevel === hierarchyLevel)
           .map((d) => d.properties.label);
 
-        return data.filter((d) => hierarchyLabels.includes(getLabel(d)));
+        return data
+          .filter((d) => dimensionLabels.includes(getLabel(d)))
+          .filter((d) => hierarchyLabels.includes(getLabel(d)));
```

Comment:
It would be better to have an index to check for inclusion here as otherwise we are looping in N^2.

```
const index = Object.fromEntries([...dimensionLabel, ...hierarchyLabels].map(l => [l, true]))
return data.filter(d => !!index[getLabel(d)])
```

Reply:
I have modified the code, so it's no longer needed 🥳